nym-vpn-cli/src/commands.rs

```diff
@@ -79,9 +85,13 @@ pub(crate) struct CliArgs {
     #[arg(long, value_parser = validate_wg_ip, requires = "enable_wireguard")]
     pub(crate) wg_ip: Option<Ipv4Addr>,
 
-    /// The IP address of the nym TUN device that wraps IP packets in sphinx packets.
-    #[arg(long, alias = "ip", value_parser = validate_ip)]
-    pub(crate) nym_ip: Option<Ipv4Addr>,
+    /// The IPv4 address of the nym TUN device that wraps IP packets in sphinx packets.
+    #[arg(long, alias = "ipv4", value_parser = validate_ipv4, requires = "nym_ipv6")]
+    pub(crate) nym_ipv4: Option<Ipv4Addr>,
+
+    /// The IPv6 address of the nym TUN device that wraps IP packets in sphinx packets.
+    #[arg(long, alias = "ipv6", value_parser = validate_ipv6, requires = "nym_ipv4")]
+    pub(crate) nym_ipv6: Option<Ipv6Addr>,
```

Comment:
I wonder, maybe we should allow not having ipv6? Although I don't know how common that is in practice to only have a ipv4 address.. :shrug:

Reply:
After pondering on it over the weekend, I think it's safe to assume that desktop or mobile machines with network stack that don't have support for IPv6 are pretty rare. Not sure how to test this, but I think the only place there might be an error would be either when we setup the IPv6 routes or the IPv6 address of the interface. In rest, we can safely assign v6 addresses that will never be used, and it would be more robust for our part to do so.
In case such error happens, I think it would be better to deal with it at that point, probably making that specific error a warning instead, since proactively trying to figure out the v6 support of a machine doesn't seem trivial.